Mathlib/Data/Nat/Digits.lean

```diff
@@ -738,111 +738,8 @@ theorem eleven_dvd_of_palindrome (p : (digits 10 n).Palindrome) (h : Even (digit
-
-open Tactic
--- failed to format: unknown constant 'term.pseudo.antiquot'
-/-- Helper function for the `norm_digits` tactic. -/ unsafe
-  def
-    eval_aux
-    ( eb : expr ) ( b : ℕ ) : expr → ℕ → instance_cache → tactic ( instance_cache × expr × expr )
-    |
-      en , n , ic
-      =>
-      do
-        let m := n / b
-          let r := n % b
-          let ( ic , er ) ← ic . ofNat r
-          let ( ic , pr ) ← norm_num.prove_lt_nat ic er eb
-          if
-            m = 0
-            then
-            do
-              let ( _ , pn0 ) ← norm_num.prove_pos ic en
-                return
-                  (
-                    ic
-                      ,
-                      q( ( [ $ ( en ) ] : List Nat ) )
-                        ,
-                        q( digits_one $ ( eb ) $ ( en ) $ ( pn0 ) $ ( pr ) )
-                    )
-            else
-            do
-              let em ← expr.of_nat q( ℕ ) m
-                let ( _ , pe ) ← norm_num.derive q( ( $ ( er ) + $ ( eb ) * $ ( em ) : ℕ ) )
-                let ( ic , el , p ) ← eval_aux em m ic
-                return
-                  (
-                    ic
-                      ,
-                      q( @ List.cons ℕ $ ( er ) $ ( el ) )
-                        ,
-                        q(
-                          digits_succ
-                            $ ( eb ) $ ( en ) $ ( em ) $ ( er ) $ ( el ) $ ( pe ) $ ( pr ) $ ( p )
-                          )
-                    )
-#align nat.norm_digits.eval_aux Nat.NormDigits.eval_aux
-
-/-- A tactic for normalizing expressions of the form `Nat.digits a b = l` where
-`a` and `b` are numerals.
-
-```
-example : Nat.digits 10 123 = [3,2,1] := by norm_num
-```
--/
-@[norm_num]
-unsafe def eval : expr → tactic (expr × expr)
-  | q(Nat.digits $(eb) $(en)) => do
-    let b ← expr.to_nat eb
-    let n ← expr.to_nat en
-    if n = 0 then return (q(([] : List ℕ)), q(Nat.digits_zero $(eb)))
-      else
-        if b = 0 then do
-          let ic ← mk_instance_cache q(ℕ)
-          let (_, pn0) ← norm_num.prove_ne_zero' ic en
-          return (q(([$(en)] : List ℕ)), q(@Nat.digits_zero_succ' $(en) $(pn0)))
-        else
-          if b = 1 then do
-            let ic ← mk_instance_cache q(ℕ)
-            let s ← simp_lemmas.add_simp simp_lemmas.mk `list.replicate
-            let (rhs, p2, _) ← simplify s [] q(List.replicate $(en) 1)
-            let p ← mk_eq_trans q(Nat.digits_one $(en)) p2
-            return (rhs, p)
-          else do
-            let ic ← mk_instance_cache q(ℕ)
-            let (_, l, p) ← eval_aux eb b en n ic
-            let p ← mk_app `` And.left [p]
-            return (l, p)
-  | _ => failed
-#align nat.norm_digits.eval Nat.NormDigits.eval
--/
-
-end NormDigits
+macro "simp_digits" : tactic =>
+  `(tactic| simp only [digits_of_two_le_of_pos, reduceLE, reduceMod, reduceDiv,
+                       zero_lt_succ, digits_zero, reduceMul, reduceAdd])
```

Comment:
It would be nicer to make this a proper simp set via `register_simp_attr`, but as far as I can tell simprocs (like `reduceLE`, `reduceMod`...) cannot be added to simp sets (they seem to just get silently ignored).